server/db.js

```diff
@@ -0,0 +1,37 @@
+const Mongoose = require('mongoose');
+
+const getMongoDBURL = (config)=>{
+	return config.get('mongodb_uri') ||
+           config.get('mongolab_uri') ||
+           'mongodb://localhost/homebrewery';
+};
+
+const disconnect = ()=>{
+	return Mongoose.close();
+};
+
+const connect = (config)=>{
+	const resolver = (resolve, reject)=>{
+		Mongoose.connect(getMongoDBURL(config),
+			{ retryWrites: false },
+			(error)=>{
+				if(error) {
+					console.error('Could not connect to a Mongo Database.');
+					console.log(error);
+					console.error('If you are running locally, make sure mongodb.exe is running.');
+					// FIXME: do we need to pass 'error' to 'reject'?
+					return reject();
+				}
+
+				return resolve();
+			});
+
+	};
+
+	return new Promise(resolver);

```

Comment:
In general we've been using await/async and `.catch()` rather than the traditional `resolve/reject` promise syntax. Could you change to that to keep it unified with our other code?

If you want, you can add a `.catch` here to handle the error, but we also use `express-async-handler` to make it a bit easier to send errors along to our middleware without it, if you can make that work.

Reply:
Refactored to use `async`/`await` in e84cd4fe8b589f687cd29746910864fe80bc57d3